### PR TITLE
:sparkles: add support to go 1.19

### PR DESCRIPTION
**Pull request by oscr**
Updates:
* go1.18/go1.17 -> go1.19
* goreleaser 1.9.2 -> 1.11.2

Related pr #2920 Update controller-runtime, kubebuilder-declarative... 
But separating the Golang update for the rest

### CODE REVIEW

**Comment by jmrodri** [comment]
@oscr you'll have to run `make generate` to fix the testdata.

**Comment by oscr** [comment]
Blocked until this is merged https://github.com/kubernetes/test-infra/pull/27395

**Comment by oscr** [comment]
@jmrodri Thank you for your advice :pray:

**Review by camilamacedo86 — APPROVED** [review]
Hi @oscr, 

Thank you for the contribution 🥇 .
I hope that you do not mind. 
I just update the title so we move forward because updating goreleaser is something that does not matter for those that use the tool. 

In the changelog which is generated automatically, we just add the changes that can impact the CLI users and those that are consumers of the lib.  

/lgtm
/approved

File: pkg/plugins/golang/v3/init.go
Comment (camilamacedo86) [inline pkg/plugins/golang/v3/init.go:39]: @oscr 

we need to revert this one to be able to move forward, see:

> failed to initialize project: unable to run pre-scaffold tasks of "base.go.kubebuilder.io/v3": go version 'go1.18.6' is incompatible because 'plugin requires go1.19 <= version < go2.0alpha1'. You can skip this check using the --skip-go-version-check flag

Also, based on the changes from 1.18 to 1.19 the solution might still be working with 1.18 so I think we do not need 
to change the min required version for now.

/hold

**Comment by oscr** [comment]
Thank you for cleaning up the pr @camilamacedo86. :rocket: I was unaware of the autogenerated changelog.

**Pull request merged**
